File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;
        Deque<Integer> q = new LinkedList<>();

        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            String command = st.nextToken();

            if (command.equals("push")) {
                q.add(Integer.parseInt(st.nextToken()));
            } else if (command.equals("pop")) {
                sb.append(q.isEmpty() ? -1 : q.poll()).append("\n");
            } else if (command.equals("size")) {
                sb.append(q.size()).append("\n");
            } else if (command.equals("empty")) {
                sb.append(q.isEmpty() ? 1 : 0).append("\n");
            } else if (command.equals("front")) {
                sb.append(q.isEmpty() ? -1 : q.peekFirst()).append("\n");
            } else if (command.equals("back")) {
                sb.append(q.isEmpty() ? -1 : q.peekLast()).append("\n");
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

n = int(sys.stdin.readline())
queue = deque()

for i in range(n):
    cmd = sys.stdin.readline().split()

    if cmd[0] == 'push':
        queue.append(cmd[1])
    elif cmd[0] == 'pop':
        if len(queue) != 0:
            print(queue.popleft())
        else:
            print(-1)
    elif cmd[0] == 'size':
        print(len(queue))
    elif cmd[0] == 'empty':
        if len(queue) != 0:
            print(0)
        else:
            print(1)
    elif cmd[0] == 'front':
        if len(queue) != 0:
            print(queue[0])
        else:
            print(-1)
    elif cmd[0] == 'back':
        if len(queue) != 0:
            print(queue[-1])
        else:
            print(-1)